sklearn/decomposition/factor_analysis.py

```diff
@@ -159,8 +164,13 @@ def fit(self, X, y=None):
 
         n_samples, n_features = X.shape
         n_components = self.n_components
-        if n_components is None:
+        if self.rotation is not None:
+            rot_components = (n_components if n_components is not None
```

Comment:
Is rot_components meant to be used? 
Otherwise n_components by default will be n_features?

Reply:
Good catch!